File: Stack/StackLinkedList.py
import sys
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None
    

class Stack:
    def __init__(self):
        self.head = None

    def isEmpty(self):
        return self.head == None
    
    def push(self,X):
        if self.isEmpty():
            self.head = X
        else:
            X.next = self.head
            self.head = X
    
    def pop(self):
        if self.isEmpty():
            return None
        else:
            X = self.head
            self.head = self.head.next
            print(X.value)
            return X

S = Stack()    
N = int(input())

for i in range(N):
    str = input().split()
    if str[0][0]=='P':
        if (str[0][1] =='U' ):
            S.push(Node(int(str[1])))
        else:
            S.pop()

File: Stack/StackLinkedList.java
package Week5.Stack;
import java.util.Scanner;

public class StackLinkedList {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        s.nextLine();
        Stack stack = new Stack();
        for (int i=0;i<n;i++) {
            String str = s.nextLine();
            String[] a = str.split(" ");
            if (a[0].equals("PU")) {
                stack.push(Integer.parseInt(a[1]));
            } else if (a[0].equals("PO")) 
                System.out.println(stack.pop()+" ");
        }
        s.close();        
    }
}

class Node {
    int key;
    Node next;

    Node() {
        this.next = null;
    }

    Node(int x) {
        this.key = x;
        this.next = null;
    }
}

class Stack {
    Node head;

    Stack() {
        this.head = null;
    }

    public boolean isEmpty() {
        return this.head == null;
    }

    public void push(int x) {
        Node q = new Node(x);
        //q.key = x;
        q.next = this.head;
        this.head = q;
    }

    public int pop() {
        if (this.isEmpty()) { throw new Error("Stack is empty");}
        int val = this.head.key;
        this.head = this.head.next;
        return val;
    }
}